lib/modules/manager/kubernetes/extract.ts

```diff
@@ -37,14 +37,22 @@ export function extractPackageFile(
   return deps.length ? { deps } : null;
 }
 
+// Comes from https://github.com/distribution/reference/blob/v0.6.0/regexp.go
+// Extracted & converted with https://go.dev/play/p/ZwF3vvRD9Rs
+const dockerImageRegexPattern = `((?:(?:(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])(?:\\.(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]))*|\\[(?:[a-fA-F0-9:]+)\\])(?::[0-9]+)?/)?[a-z0-9]+(?:(?:[._]|__|[-]+)[a-z0-9]+)*(?:/[a-z0-9]+(?:(?:[._]|__|[-]+)[a-z0-9]+)*)*)(?::([A-Za-z0-9][A-Za-z0-9.-]{0,127}))?(?:@([A-Za-z][A-Za-z0-9]*(?:[-_+.][A-Za-z][A-Za-z0-9]*)*[:][0-9a-fA-F]{32,}))?`;
```

Comment:
@zharinov do you think we can simplify this?

Reply:
Problems:
- we risk making a mistake 
- if the optimization is manual then we need to repeat it anyone the source changes